### ISSUE TITLE
no-top-level-browser-globals rule false positive inside $state generic

### ISSUE THREAD

**Issue by that-richan** [sveltejs/eslint-plugin-svelte#1361]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.35.0

### What version of `eslint-plugin-svelte` are you using?

3.12.3

### What did you do?

```svelte
<script lang="ts">
	const element: HTMLElement | null = null; // fine.
	const stateElement = $state<HTMLElement | null>(null); // Unexpected top-level browser global variable "HTMLElement". (eslint svelte/no-top-level-browser-globals)
</script>
```

### What did you expect to happen?

No warning/error on the usage of the `HTMLElement` type.

### What actually happened?

An error.

### Link to **GitHub Repo** with Minimal Reproducible Example

https://github.com/that-richan/no-top-level-browser-globals-rule-svelte-state

And run: `npx eslint .`

**Comment by that-richan** [comment]
Added example. @ota-meshi
